keps/sig-storage/3751-volume-attributes-class/README.md

```diff
@@ -439,31 +439,29 @@ Operation metrics from [csiOperationsLatencyMetric](https://github.com/kubernete
 
 While a VolumeAttributesClass is referenced by any PVC, we will prevent the object from being deleted by adding a finalizer([reference](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/storage/storageobjectinuseprotection/admission.go)). 
 
-The VACObjectInUseProtection admission controller sets the finalizer on all VolumeAttributesClasses. VACProtectionController removes the finalizer when it's not referenced. This prevents users from deleting a VolumeAttributesClass that's used by a PVC.
+The external-provisioner and external-resizer set/remove the finalizer on all VolumeAttributesClasses. This prevents users from deleting a VolumeAttributesClass that's used by a PVC.
 
 There are a few conditions that will trigger add/remove pvc finalizers in the VolumeAttributesClass:
 
 
 1. PVC created with a VolumeAttributesClass
 
-    The **VACObjectInUseProtection admission controller**:
-    * Check if the VolumeAttributesClass exists. If not, the PVC will enter the INPROGRESS state because we do not want to impose ordering on object creation
+    The **external-provisioner**:
+    * Check if the VolumeAttributesClass exists. If not, the PVC will enter the PENDING state because we do not want to impose ordering on object creation
     * Check if this VolumeAttributesClass already has a protection finalizer
     * Add the finalizer to the VolumeAttributesClass if there is none 
 2. PVC created with a VolumeAttributesClass being deleted
-    The **VACObjectInUseProtection admission controller**:
+    The **external-provisioner**:
     * Check VolumeAttributesClass is being deleted and PVC creation failed
 3. PVC updated to a different VolumeAttributesClass
-    * The **VACProtectionController** will remove PVC finalizer in previous VolumeAttributesClass if after listing all the PVCs and confirmed that this PVC is the last one that is consuming the previous VolumeAttributesClass
-
-    The **VACObjectInUseProtection admission controller**:
+    * The **external-resizer** will **not** remove PVC finalizer in previous VolumeAttributesClass if after listing all the PVCs via Listers and confirmed that this PVC is the last one that is consuming the previous VolumeAttributesClass
```

Comment:
Why do we change "will remove PVC finalizer" to "will **not** remove PVC finalizer" if no other PVC is using the previous VAC?  In which case will that finalizer be removed?

Reply:
Because we only want the provisioner to remove PVC finalizer and leave the resizer out of this to avoid more race conditions, there is an explanation added at the end of this session.

Reply:
Why does the resizer need to List all PVCs if it's not going to handle finalizer removal?

Also I wonder if provisioner needs to periodically check for finalizer removal conditions. It seems very easy to run into a scenario where we will not cleanup the finalizer properly:
1. Create PVC with VAC1
2. Change PVC to VAC2
3. Try to delete VAC1.

Reply:
Oh this should be removed.